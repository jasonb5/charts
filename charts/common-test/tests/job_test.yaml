suite: test job
values:
  - values/common.yaml
templates:
  - app.yaml
chart:
  appVersion: 0.1.0
tests:
  - it: should overwrite defaults
    set:
      workload: job

      activeDeadlineSeconds: 10
      backoffLimit: 10
      backoffLimitPerIndex: 2
      completionMode: Indexed
      completions: 10
      maxFailedIndexes: 2
      parallelism: 10
      podFailurePolicy:
        rules:
        - action: FailJob
          onExitCodes:
            containerName: test
            operator: "In"
            values:
            - 0
      podReplacementPolicy: TerminatingOrFailed
      ttlSecondsAfterFinished: 20
    asserts:
      - matchSnapshot:
          path: spec
        documentIndex: 0
      - hasDocuments:
          count: 1
  - it: should pass
    set:
      workload: job
    asserts:
      - matchSnapshot: {}
        documentIndex: 0
      - hasDocuments:
          count: 1
