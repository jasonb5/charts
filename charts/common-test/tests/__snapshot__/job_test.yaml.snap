should overwrite defaults:
  1: |
    activeDeadlineSeconds: 10
    backoffLimit: 10
    backoffLimitPerIndex: 2
    completionMode: Indexed
    completions: 10
    maxFailedIndexes: 2
    parallelism: 10
    podFailurePolicy:
      rules:
        - action: FailJob
          onExitCodes:
            containerName: test
            operator: In
            values:
              - 0
    podReplacementPolicy: TerminatingOrFailed
    template:
      spec:
        activeDeadlineSeconds: 10
        containers:
          - image: busybox:0.1.0
            name: common-test
    ttlSecondsAfterFinished: 20
should pass:
  1: |
    apiVersion: v1
    kind: Job
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: common-test
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: common-test-0.1.0
      name: RELEASE-NAME-common-test
      namespace: NAMESPACE
    spec:
      template:
        spec:
          containers:
            - image: busybox:0.1.0
              name: common-test
